lxd/device/cdi/log.go

```diff
@@ -0,0 +1,40 @@
+package cdi
+
+import (
+	"fmt"
+
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// CDILogger reuses LXD's shared logger to log the internal operations of the CDI spec generator.
+type CDILogger struct{}
```

Comment:
Can you add a NewCDILogger function that takes an existing logger as an argument, then that way we can pass in an existing contextual logger (that has instance and device specific context) rather than use the global logger and get messages without any context as to which instance device it relates to.

Reply:
fixed